IPs/RSKIP197.md

```diff
@@ -0,0 +1,78 @@
+# Error Handling for Precompiled Contracts
+
+|RSKIP          |197           |
+| :------------ |:-------------|
+|**Title**      |Error Handling for Precompiled Contracts |
+|**Created**    |15-12-2020 |
+|**Author**     |FJ |
+|**Purpose**    |USa |
+|**Layer**      |Core |
+|**Complexity** |2 |
+|**Status**     |Accepted |
+
+## Abstract
+
+This RSKIP specifies a new expected behavior for a precompiled contract call: any expected failed call pushes a zero into the stack and reverts the state before the execution.
+
+## Motivation
+
+Provides a mechanisim to handle errors for precompiled calls by checking the returned state. Before Iris hardfork the transaction wasn't reverted and consumed all the remaining gas from the entire call.
+
+Suppose that there is a Contract A that, somewhere, calls a Precompiled contract B with invalid data (or something that produces an expected failure), it's desirable to throw a kind "exception" that gives the chance to the caller (A) to handle and use it for it's own pouposes, instead of failing the entire execution.
+
+## Specification
+
+When `blockNumber >= IRIS_HF`, during a precompiled contract execution any expected failure should:
+1. Revert the execution state.
+2. Push a ZERO into the stack.
```

Comment:
This specification is unclear. The ZERO or ONE is pushed in the stack according to the CALL semantics. Are you changing the CALL semantics? Are yo making a CALL to a pre-compiler respect this semantics ?